workflow/cron/controller.go

```diff
@@ -94,7 +94,22 @@ func (cc *Controller) Run(ctx context.Context) {
 	defer cc.cron.Stop()
 
 	go cc.cronWfInformer.Informer().Run(ctx.Done())
-	go wait.Until(cc.syncAll, 10*time.Second, ctx.Done())
+	go func() {
+		// To minimize syncAll and scheduled crons stepping over each other, ensure that syncAll runs every 10 seconds
+		// starting on a x5 second mark (e.g., 15, 35, etc.). Since crons are guaranteed to run on a 00 second mark, this
+		// makes sure that there is always a 5 second time separation between syncAll opeartions and any scheduled cron
+		_, _, sec := time.Now().Clock()
+		singleSec := sec % 10
+		var toWait time.Duration
+		if singleSec <= 5 {
+			toWait = time.Duration(5-singleSec) * time.Second
```

Comment:
this is pretty reliant on time.Now() - can we try to use something more deterministic?

Reply:
This has been removed... no longer required

Reply:
And this was unrelated to https://github.com/argoproj/argo/pull/4638#discussion_r537648603. In fact, the purpose of this code was to make something stochastic more deterministic